File: readability.c

#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    int letters = 0;
    int words = 1;
    int sentences = 0;
    float index = 0;
    string text = get_string("Text:\n");
    
    
    for (int i = 0; i < strlen(text); i++)
    {   
       if isalpha (text[i]) 
       { 
           letters = letters +1;
       }
       if isspace (text[i])
       {
           words = words +1;
       }
       if (text[i] =='!' || text[i] =='.' || text[i] =='?')
       {
           sentences = sentences +1;
       }
    }
    
index = .0588*((float)letters/((float)words/(float)100))- .296*((float)sentences/((float)words/(float)100)) - 15.8;
int grade = round(index);

if (grade < 1)
{
printf("Before Grade 1\n");
}
else if (grade >= 16)
{
    printf("Grade 16+\n");
}
else
{
    printf("Grade %i\n", grade);
}
}

File: readability.py
from cs50 import get_string

# get user input
text = get_string("Text: ")

# initialize variables
letters = 0 
words = 1
sentences = 0

# iterate through text
for c in text:
    a = ord(c)
    
    # count letters 
    if a > 64 and a < 90 or a > 96 and a < 122:
        letters += 1
    
    # count words
    elif (a == 32):
        words += 1
    
    # count sentences
    elif a == 46 or a == 63 or a == 33:
        sentences += 1
        
# calculate Coleman-Liau
L = letters / (words / 100)
S = sentences / (words / 100)
CL = .0588 * L - .296 * S - 15.8

# round and return grade level
grade = round(CL)
if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade} ")